homeassistant/components/reolink/host.py

```diff
@@ -317,11 +335,22 @@ async def disconnect(self):
                 str(err),
             )
 
-    async def _async_start_long_polling(self):
+    async def _async_start_long_polling(self, initial=False):
         """Start ONVIF long polling task."""
         if self._long_poll_task is None:
             try:
                 await self._api.subscribe(sub_type=SubType.long_poll)
+            except NotSupportedError as err:
+                if initial:
+                    raise err
+                # make sure the long_poll_task is always created to try again later
+                if not self._lost_subscription:
+                    self._lost_subscription = True
+                    _LOGGER.error(
+                        "Reolink %s event long polling subscription lost: %s",
+                        self._api.nvr_name,
+                        str(err),
```

Comment:
No need to copy to string. Exceptions have a magic string method already.

Reply:
Thanks for letting me know, I have cleaned this up in this PR: https://github.com/home-assistant/core/pull/100718